pmsoService/product_flow/features/deliverytrip/app/handlers.py

```
@@ -1,3 +1,346 @@
+        sale_order_item_ids = [str(i.sale_order_item_id) for i in by_sale]
+        count_belong = delivery_items_repo.count_sale_order_items_belonging_to_order(
+            sale_order_id, sale_order_item_ids
+        )
+        validate_items_belong_to_delivery_trip(count_belong, len(sale_order_item_ids))
+
+        # Basic create validation equivalence
+        from product_flow.features.deliverytrip.app.commands import CreateDeliveryItem
+
+        for it in by_sale:
+            if it.item_kind is None:
+                raise ValueError("item_kind là bắt buộc khi dùng sale_order_item_id")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)